java/java.hints/src/org/netbeans/modules/java/hints/errors/EnablePreviewFeature.java

```diff
@@ -0,0 +1,415 @@
+                                is.close();
+                            }
+                        }
+                    }
+                    return ep;
+                }
+            });
+        } catch (MutexException ex) {
+            return null;
+        }
+    }
+
+    private boolean isMavenProject(Project prj) {
+
+        FileObject pom = prj.getProjectDirectory().getFileObject("pom.xml");
+        return (pom != null) && pom.isValid();
+
+    }
+
+    private boolean isAntProject(Project prj) {
+
+        FileObject buildFile = prj.getProjectDirectory().getFileObject("build.xml");
+        return (buildFile != null) && buildFile.isValid();
+
+    }
+
+    PROJ_TYPES getProjectType(Project prj) {
+
+        if (isAntProject(prj)) {
+            return PROJ_TYPES.ANT;
+        }
+        if (isMavenProject(prj)) {
+            return PROJ_TYPES.MAVEN;
+        } else {
+            return null;
+        }
+
+    }
+
+    private static class AddMvnCompilerPluginForEnablePreview implements ModelOperation<POMModel> {
+
+        private static final String MAVEN_COMPILER_GROUP_ID = "org.apache.maven.plugins"; // NOI18N
+        private static final String MAVEN_COMPILER_ARTIFACT_ID = "maven-compiler-plugin"; // NOI18N
+        private static final String COMPILER_ID_PROPERTY = "compilerId"; // NOI18N
+        private static final String COMPILER_ARG = "compilerArgs"; // NOI18N
+        private static final String MAVEN_COMPILER_VERSION = "3.3" ; // NOI18N
+        private static final String ARG = "arg";// NOI18N
+        private POMComponentFactory factory;
+
+        @Override
+        public void performOperation(final POMModel model) {
+            factory = model.getFactory();
+            org.netbeans.modules.maven.model.pom.Project proj = model.getProject();
+            Build build = model.getProject().getBuild();
+            if (build == null) {
+                build = factory.createBuild();
+                proj.setBuild(build);
+            }
+
+            Plugin oldPlugin = searchMavenCompilerPlugin(build);
+            if (oldPlugin == null) {
+                build.addPlugin(createMavenEclipseCompilerPlugin());
+            } else {
+
+                Plugin newPlugin = updateMavenEclipseCompilerPlugin(oldPlugin);
+
+                build.removePlugin(oldPlugin);
+                build.addPlugin(newPlugin);
+            }
+        }
+
+        private Plugin searchMavenCompilerPlugin(final Build build) {
+            List<Plugin> plugins = build.getPlugins();
+            if (plugins != null) {
+                for (Plugin plugin : plugins) {
+                    if (MAVEN_COMPILER_GROUP_ID.equals(plugin.getGroupId())
+                            && MAVEN_COMPILER_ARTIFACT_ID.equals(plugin.getArtifactId())) {
+                        return plugin;
+                    }
+                }
+            }
+            return null;
+        }
+
+        private Plugin createMavenEclipseCompilerPlugin() {
```

Comment:
Why "eclipse"?

Reply:
My mistake i should had corrected the function name.
The code was copied from https://github.com/apache/netbeans/blob/master/groovy/maven.groovy/src/org/netbeans/modules/maven/groovy/extender/AddMavenCompilerPlugin.java